Libplanet/Store/DefaultStore.cs

```diff
@@ -251,8 +251,13 @@ public override bool DeleteIndex(Guid chainId, HashDigest<SHA256> hash)
             LiteCollection<HashDoc> srcColl = IndexCollection(sourceChainId);
             LiteCollection<HashDoc> destColl = IndexCollection(destinationChainId);
 
-            destColl.InsertBulk(srcColl.FindAll().TakeWhile(i => !i.Hash.Equals(branchPoint)));
-            AppendIndex(destinationChainId, branchPoint);
+            var genesisHash = IterateIndexes(sourceChainId, 0, 1).First();
+            destColl.InsertBulk(srcColl.FindAll().TakeWhile(i => !i.Hash.Equals(branchPoint))
```

Comment:
1. Doesn't this op relate with the below condition(`!branchPoint.Equals(genesisHash)`)?
2. I guess `genesis` is only one and first hash, so we can use `Skip(1)` instead of `SkipWhile()`, is it right?

Reply:
1. It seems I added the condition to except the index of the genesis block. If it doesn't check, there will be two same genesis block's hash on the chain.
2. You are right :grin: I will amend.